src/bidiMapper/utils/log.ts

```diff
@@ -15,16 +15,19 @@
  */
 export function log(type: string): (...message: any[]) => void {
   return (...messages: any[]) => {
-    const elementId = type + '_log';
+    console.log.apply(null, [type].concat(messages));
 
-    if (!window.document.getElementById(elementId)) {
-      window.document.documentElement.innerHTML += `<h3>${type}:</h3><pre id='${elementId}'></pre>`;
-      window.document.getElementById(elementId);
-    }
+      // If run in browser, add debug message to the page.
+    if (typeof window !== 'undefined' && window && window.document && window.document.documentElement) {
```

Comment:
This could be simplified using [optional chaining](https://v8.dev/features/optional-chaining)

Reply:
`typeof window !== 'undefined'` is still needed, but the next chain is simplified. Thanks!